examples/s/d/nn/_project/aggregate_surfaces.py

```diff
@@ -0,0 +1,236 @@
+"""Use fmu-dataio for aggregated surfaces created by an aggregation service."""
+
+from pathlib import Path
+import logging
+
+import yaml
+import numpy as np
+
+import xtgeo
+import fmu.dataio
+
+
+def main():
+    """Aggregate one surface across X realizations from the example case and store the
+    results. In this example, we emulate that fmu-dataio is called by another service.
+
+    Two contexts are demonstrated:
+
+        1) We are in a classical FMU setting, running this aggregation as a stand-alone
+           Python script (directly, or wrapped in an ERT workflow). In this context,
+           we want the resulting files to be stored to disk within the existing case. In
+           this context, fmu-dataio is responsible for storing the results.
+        2) We are in a cloud service. In this context, we don't want to store anything
+           on disk, as the results are to be pushed to other storage, i.e. Sumo. Hence,
+           we want fmu-dataio to provide us with the generated metadata only. In this
+           context, the service itself is responsible for storing the results.
+
+    Note that this example is showing the usage of fmu-dataio, it is not to be seen as
+    an example for the actual aggregation. The aggregation service shown here is
+    simplistic and its sole purpose is to facilitate the fmu-dataio example.
+
+    """
+
+    # First we get the input data (the individual surfaces from each realization), which
+    # we assume are stored in classical FMU style on /scratch disk folder structure.
+
+    # IRL, these variables would typically be arguments to the aggregation script.
+    casepath = Path("../xcase/").resolve()
+    iter_name = "iter-0"
+    relative_path = (
+        "share/results/maps/topvolantis--ds_extract_geogrid.gri"  # exists in all reals
+    )
+    realization_ids = _get_realization_ids(casepath)
+
+    # gather source surfaces and their associated metadata
+    source_surfaces, source_metadata = _get_source_surfaces_from_disk(
+        casepath=casepath,
+        iter_name=iter_name,
+        realization_ids=realization_ids,
+        relative_path=relative_path,
+    )
+
+    # These are the operations we want to do
+    operations = ["mean", "min", "max", "std"]
+
+    # This is the ID we assign to this set of aggregations
+    aggregation_id = "something_very_unique"  # IRL this will usually be a uuid
+
+    # We aggregate these source surfaces and collect results in list of dictionaries
+    aggregations = []
+
+    # Initialize an AggregatedData object for this set of aggregations
+    exp = fmu.dataio.AggregatedData(
+        source_metadata=source_metadata,
+        aggregation_id=aggregation_id,
```

Comment:
Just wording, but would "aggrexp" be a better name than "exp" here just to not confuse it with an ExportData instance?